app/src/main/java/com/epfl/drawyourpath/mainpage/MainActivity.kt

```diff
@@ -65,9 +66,8 @@ class MainActivity : AppCompatActivity() {
         // Create an instance of your database
         val database: Database = FirebaseDatabase()
 
-        // Create an instance of FriendsFragmentFactory and set it as the fragment factory
-        val friendsFragmentFactory = FriendsFragmentFactory(database)
-        supportFragmentManager.fragmentFactory = friendsFragmentFactory
+        // Set the fragment factory to the custom fragment factory to help initialise the friends fragment with the database dependency
```

Comment:
Why do you instantiate the friend fragment in the main activity and not of the launch of the friend fragment ?

Reply:
Its not the FriendsFragment that is instanciated I think, but rather the factory that's used to create it.